### PR TITLE
feat(rust): check controller's identity id when creating secure channel

### PR DESCRIPTION
**Pull request by adrianbenavides**
https://github.com/build-trust/codebase/issues/330
https://github.com/build-trust/codebase/issues/340

The controller's identity can be loaded in three different ways:

1. ockam binary: the contents from `implementations/rust/ockam/ockam_api/static/controller.id` will be embedded into the binary and will be used by default
2. dev (cargo run ...): the contents from `implementations/rust/ockam/ockam_api/static/controller.id` will be used also in this case, but you can change its contents to try different id's
3. `OCKAM_CONTROLLER_IDENTITY_ID` env var: if it's present, its contents will be used and will have priority over the previous methods

## How to test

If running minikube on `ockam-cloud`, copy the `./minikube/mnt/storage/controller/identity.id` contents. We'll use it to set the `OCKAM_CONTROLLER_IDENTITY_ID` env var.

```bash
OCKAM_CONTROLLER_IDENTITY_ID={identity.id-contents} ockam node create default
ockam enroll --auth0 $CONTROLLER # this method will use a secure channel using the controller's identity id
```

Alternatively, you can copy the contents into the `implementations/rust/ockam/ockam_api/static/controller.id` file.

## Other changes

- Add a new registry entry into the node to cache secure channels and avoid the [resource leak](https://github.com/build-trust/ockam/issues/3159) on deleting them
- Remove cloud tests as they are no longer useful. They were testing against a fake server and were useful at the beginning to test how it all worked, but they are not testing anything really and we'll shortly have new integration tests for this

### CODE REVIEW

File: implementations/rust/ockam/ockam_api/src/nodes/registry.rs [hunk 1]
```diff
@@ -1,10 +1,37 @@
...
+        &self.addr
+    }
+}
+
+#[derive(Default, Hash, Ord, PartialOrd, Eq, PartialEq)]
+pub(crate) struct IdentityRouteKey(Vec<u8>);
+
+impl IdentityRouteKey {
+    pub(crate) async fn new(identity: &Identity<Vault>, route: &Route) -> Result<Self> {
+        let mut key = identity.export().await?;
```
Comment (SanjoDeundiak) [inline implementations/rust/ockam/ockam_api/src/nodes/registry.rs:29]: Could you please elaborate why identity should be present in the key? NodeManager can't have more than one

Comment (adrianbenavides) [reply implementations/rust/ockam/ockam_api/src/nodes/registry.rs:29]: Not needed! This is a remnant of a wrong approach I did when I started to work on this PR. I'll leave just the `route` as a key. The idea is to reuse the secure channels that are used between node and orchestrator (i.e. controller + project nodes).

Comment (adrianbenavides) [reply implementations/rust/ockam/ockam_api/src/nodes/registry.rs:29]: I was now reviewing the open issues we have and it seems we want to support multiple identities on a node: https://github.com/build-trust/ockam/issues/3186

So I guess I should revert and use `IdentityRouteKey` as a key.

**Review by spacekookie — DISMISSED** [review]
LGTM

**Review by mrinalwadhwa — DISMISSED** [review]

**Review by mrinalwadhwa — APPROVED** [review]

**Pull request merged**